models/Element/Service.php

```diff
@@ -1070,40 +1070,30 @@ public static function getValidKey(string $key, string $type): string
             'type' => $type,
         ]);
         \Pimcore::getEventDispatcher()->dispatch($event, SystemEvents::SERVICE_PRE_GET_VALID_KEY);
-        $key = $event->getArgument('key');
-        $key = trim($key);
-
-        // replace all control/unassigned and invalid characters
-        $key = preg_replace('/[^\PCc^\PCn^\PCs]/u', '', $key);
```

Comment:
This one seems broken, as I [already elaborated here](https://github.com/pimcore/pimcore/pull/13382#issuecomment-1367473344).

- `[^` means: not one of the following, repeating the `^` doesn't make sense except you wanna match it
- I don't think `\PCc` (etc.) works. I think it's seen as a `\PC` followed by a literal `c` and should be `\P{Cc}` instead.

So this regex **should** match everything that's **not** a `\P{Cc}`, `\P{Cf}`, `\P{Co}`, `\P{Cs}`, `\P{Cn}` (that's all `\PC`'s, which is three times in the list), or `^` (which is two times in the list) or `c`, `n`, `s` and replace it with nothing.
But, as far as I can see, it's not doing that. It seems that there's some kind of "security mechanism" which reads this as just `[^\PC]` - which is good, but still means it matches more than expected (which is probably also good, because I'm not sure why we *don't* wanna remove the "invisible formatting indicators" (`\p{Cf}`) or "code points reserved for private use" (`\p{Co}`)).

So I think we should instead use `/[\p{Cc}\p{Cn}\p{Cs}]/u` (or even just `/\p{C}/u`, because I don't see why we don't wanna remove format & private use characters).

Reply:
Thank you for cleaning and clearing this up. I am totally with you and we should remove all sorts chars from the `{C}` group so `/\p{C}/u`  is the way to go.